server/datastore/mysql/queries.go

```diff
@@ -493,3 +499,30 @@ func (ds *Datastore) ListScheduledQueriesForAgents(ctx context.Context, teamID *
 
 	return results, nil
 }
+
+func (ds *Datastore) DeleteAllResultsForQuery(ctx context.Context, queryId uint) error {
+	deleteStmt := fmt.Sprintf(`DELETE FROM %s WHERE query_id = ?`, queryResultsTable.name)
+	result, err := ds.writer(ctx).ExecContext(ctx, deleteStmt, queryId)
+	if err != nil {
+		return ctxerr.Wrapf(ctx, err, "delete %s by query id", queryResultsTable)
+	}
+	rows, _ := result.RowsAffected()
+	if rows < 1 {
+		return ctxerr.Wrap(ctx, notFound(queryResultsTable.name).WithID(queryId))
+	}
+	return nil
+}
+
+// DeleteAllResultsForQueryByName is needed for deleting query results when modifying specs, as we don't have the query ID in that context.
+func (ds *Datastore) DeleteAllResultsForQueryByName(ctx context.Context, queryName string) error {
+	deleteStmt := `DELETE query_results FROM query_results LEFT JOIN queries ON query_results.query_id = queries.id WHERE queries.name = ?;`
+	result, err := ds.writer(ctx).ExecContext(ctx, deleteStmt, queryName)
+	if err != nil {
+		return ctxerr.Wrapf(ctx, err, "delete %s by query name", queryResultsTable)
+	}
+	rows, _ := result.RowsAffected()
+	if rows < 1 {
+		return ctxerr.Wrap(ctx, notFound(queryResultsTable.name).WithName(queryName))
```

Comment:
Given how this method is used it seems it shouldn't fail if there are no results stored for this query.

Reply:
This comment still applies.

If you edit a query (say, change logging to something else like "differential") and the query does not have results you don't want the request to fail, so this method should not fail if no rows were affected.

Reply:
Forgot to remove this function. In another comment you pointed out we could just delete by query ID, so this isn't actually needed. Removing